Employees.java

```diff
@@ -0,0 +1,269 @@
+	        //comparing fetched and passed id
+	        if(checkId.equals(id)) {
+	        	return employeeObject;
+	        }
+	       return null;		
+		}
+		
+		@GET
+		@Path("name/{name}")
+		@Produces(MediaType.APPLICATION_JSON)
+		/**
+		 * method that calls checkName to retrieve employee details using the given name
+		 * @param name
+		 * @return JSONObject
+		 */
+		public JSONObject getEmployeeByName(@PathParam("name")String name) {
+			
+			
+			JSONParser jsonParser = new JSONParser();
+	        try (FileReader reader = new FileReader("employees.json"))
+	        {
+	            //Read JSON file
+	            Object obj = jsonParser.parse(reader);
+	            //storing data in json array
+	            JSONArray employeeList = (JSONArray) obj; 
+	            //json object to return resultant employee
+	            JSONObject resultantEmployee;
+	           //Iterate over employee array
+	            for(Object ob : employeeList) {
+	            	//to check each element's name using method checkName
+	        	    resultantEmployee = checkName((JSONObject)ob , name);
+	        	    if(resultantEmployee != null){
+	        		   return resultantEmployee;
+	        	   }
+	           }
+	        } catch (FileNotFoundException e) {
+	            e.printStackTrace();
+	        } catch (IOException e) {
+	            e.printStackTrace();
+	        } catch (ParseException e) {
+	            e.printStackTrace();
+	        }
+	        return null;
+		}
+
+		/**
+		 * method that checks and returns details of employee of the passed name
+		 * @param employee - object of json
+		 * @param name
+		 * @return JSONObject
+		 */
+		private JSONObject checkName(JSONObject employee, String name) {
+			
+			
+			JSONObject employeeObject = (JSONObject) employee.get("employee");
+	       //Get employee id
+	        String id = (String) employeeObject.get("id");
+	        //Get employee name
+	        String checkName = (String) employeeObject.get("Name");
+	       //comparing fetched and passed name
+	        if(checkName.equals(name)){
+	        	return employeeObject;
+	        }
+	        return null;
+		}
+	   
+		@DELETE
+		@Path("deleteEmp/{id}")
+		/**
+		 * method that deletes details of employee from the file 
+		 * @param id-id of employee to be deleted
+		 * @return JSONArray
+		 */
+		public JSONArray delEmployeeById(@PathParam("id")String id) {
+	    	
+			JSONParser jsonParser = new JSONParser();
+	        try (FileReader reader = new FileReader("employees.json")) {
+	           
+	        	//Read JSON file
+	            Object obj = jsonParser.parse(reader);
+	            //json array to store details
+	            JSONArray employeeList = (JSONArray) obj;
+	            //loop to iterate over the list to find the employee id to be deleted
+	            for(int i = 0; i < employeeList.size(); i++) {
+	            	//json object to get  employee 
+	            	JSONObject employeeObject = (JSONObject) employeeList.get(i);

```

Comment:
Instead of doing so many class cast operations all through the code, write common methods that will extract the information in the required type. You can also create an internal representation of an Employee object itself, instead of directly working with `JSONObject` everywhere in the code. For example: you can have a method `private Employee extractEmployee(JSONObject json)` that will abstract the conversion (and casting) of JSON into your custom object.